packages/catalog-process/src/services/catalogService.ts

```diff
@@ -385,20 +485,245 @@ export const catalogService = {
       agreementApprovalPolicy: seed.agreementApprovalPolicy,
     };
 
-    const filteredDescriptor = eservice.data.descriptors.filter(
-      (d) => d.id !== descriptorId
+    const filteredDescriptor = eService.data.descriptors.filter(
+      (d: Descriptor) => d.id !== descriptorId
     );
 
     const updatedEService = {
-      ...eservice,
+      ...eService,
       descriptor: [...filteredDescriptor, updatedDescriptor],
     };
 
     await eventRepository.createEvent({
       streamId: eServiceId,
-      version: eservice.metadata.version,
-      type: "UpdateDraftDescriptor",
+      version: eService.metadata.version,
+      type: "CatalogItemUpdated",
       data: updatedEService,
     });
   },
+
+  async publishDescriptor(
+    eServiceId: string,
+    descriptorId: string,
+    authData: AuthData
+  ): Promise<void> {
+    logger.info(
+      `Publishing Descriptor $descriptorId of EService ${eServiceId}`
+    );
+
+    const eService = await retrieveEService(eServiceId);
+    assertRequesterAllowed(eService.data.producerId, authData.organizationId);
+
+    const descriptor = await retrieveDescriptor(descriptorId, eService);
+    assertDescriptorIsDraft(descriptor);
```

Comment:
doesn't `retrieveDescriptor` already retrieve only draft descriptors? if so, then is `assertDescriptorIsDraft` useful? (also, if so, maybe rename to `retrieveDraftDescriptor`)